unraid_mcp/core/exceptions.py

```
@@ -19,36 +23,26 @@ class ToolError(FastMCPToolError):
+        tool_name: The tool name for error messages (e.g., "docker", "vm").
+        action: The current action being executed.
+        logger: The logger instance to use for error logging.
     """
-
-    pass
+    try:
+        yield
+    except ToolError:
+        raise
+    except Exception as e:
+        logger.error(f"Error in unraid_{tool_name} action={action}: {e}", exc_info=True)
```

Ruff (0.15.1)

[warning] Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)